### PR TITLE
feat(storybook): add SplitButton stories for visual review

### LINKED ISSUE #17040: Storybook: Add stories for `SplitButton` component
### Is your feature request related to a problem? Please describe

The new `SplitButton` component (#16033) has no Storybook stories, making it difficult to review and test its visual states (primary, secondary, danger types, different selection modes, etc.)

### Describe the solution you'd like

Add comprehensive Storybook stories for `SplitButton` that demonstrate:

- All button types (primary, secondary, danger)
- Different states (no selection, single selection, multi-selection, in progress, disabled)
- Icon variations
- Edge cases (empty options, long labels, custom labels)

### Describe alternatives you've considered

N/A

### Additional context

Related to PR #16033. Stories needed to identify visual issues before merge.

/cc @dgolovin

### PR DESCRIPTION
**Pull request by vancura**
Adds comprehensive Storybook stories for the SplitButton component introduced in #16033, enabling visual review and testing of all component states and variants.

The stories cover:

- Button types (primary, secondary, danger)
- States (ready, no selection, in progress, disabled, all options disabled, empty)
- Icon variations (button icons, option icons, both)
- Multi-select mode with checkmarks
- Edge cases (single option, long labels, custom labels, tooltips)
- Real-world interactive examples (merge strategy, deploy target, container start mode)

### Screenshot / video of UI

CleanShot 2026-04-10 at 14 48 19@2x

CleanShot 2026-04-10 at 14 48 55@2x

CleanShot 2026-04-10 at 14 49 12@2x

CleanShot 2026-04-10 at 14 49 35@2x

### What issues does this PR fix or reference?

- Related to #16033 - provides visual testing infrastructure to identify and fix styling issues before merge.
- Resolves #17040

### How to test this PR?

1. Check out the branch and run `pnpm --filter storybook dev`
2. Navigate to `Button` → `SplitButton` in the Storybook sidebar
3. Review the grouped stories (`Button Types`, `States`, `With Icons`, `Multi-select`, `Edge Cases`)
4. Test the interactive examples (`Merge Strategy`, `Deploy Target`, `Container Start Mode`)
5. Verify all button types render correctly across themes
6. Check that strokes appear correctly in dark and high-contrast themes

- [ ] Tests are covering the bug fix or the new feature (N/A - this adds visual test stories)

/cc @dgolovin 

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: packages/ui/src/lib/button/SplitButton.svelte [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+// Dropdown can be opened even if selected option is disabled (to see/change options)
+const isDropdownDisabled = $derived(disabled || isEmpty);
+// Show "no action" mode when options exist but none are actionable
+const isNoActionMode = $derived(!isEmpty && !hasActionableOption);
+
+function handleEscape({ key }: KeyboardEvent): void {
+  if (key === 'Escape') {
+    showDropdown = false;
+  }
+}
```
[the remark this answers is not in the document]
Comment (vancura) [reply packages/ui/src/lib/button/SplitButton.svelte:87]: Good tip @dgolovin

[the remark this answers is not in the document]
Comment (vancura) [reply packages/ui/src/lib/button/SplitButton.svelte:87]: The nitpick above is also quite useful.

**Comment by vancura** [comment]
@dgolovin Please take it from here. Take a look what the `SplitButton` Storybook stories revealed, there's a couple of issues to fix. We should keep these Storybook changes around, but they can't be merged right now, as they need the actual `SplitButton` to be available too.
